src/core/diags/factory_diags.cpp

```diff
@@ -676,8 +721,13 @@ Error Diags::ProcessRadio(uint8_t aArgsLength, char *aArgs[])
 
         SuccessOrExit(error = RadioReceive());
 
-        receiveConfig.mIsEnabled = true;
-        mReceiveConfig           = receiveConfig;
+        mReceiveConfig.mIsEnabled      = true;
```

Comment:
I think the original way of using '=' operator to assign the struct will work. (It will do an assignment for every field) Why do we change that?

Reply:
Because the content of the `receiveConfig` will be cleared when defining it. If we continue to use `=`, the `mFilterAddress` and `mIsFilterEnabled` will be cleared.